src/clone/clone-utils.php

```diff
@@ -0,0 +1,179 @@
+		EE::error( 'Unable to sync certs.' );
+	}
+}
+
+function copy_site_files( Site $source, Site $destination, string $sync_type ) {
+	$exclude            = '--exclude \'/wp-config.php\'';
+	$source_public_path = str_replace( '/var/www/htdocs', '', $source->site_details['site_container_fs_path'] );
+	$uploads_path       = $source_public_path . '/wp-content/uploads';
+	$uploads_path_share = '/shared/wp-content/uploads';
+
+	$source_dir      = $source->get_site_root_dir();
+	$destination_dir = $destination->get_site_root_dir();
+
+	if ( $sync_type === 'files' ) {
+		$exclude .= ' --exclude \'' . $uploads_path . '\'';
+		$exclude .= ' --exclude \'' . $uploads_path_share . '\'';
+	} elseif ( $sync_type === 'uploads' ) {
+		$source_dir      .= $uploads_path;
+		$destination_dir .= $uploads_path;
+	} elseif ( $sync_type !== 'all' ) {
+		EE::error( 'Unknown sync_type: ' . $sync_type );
+	}
+
+	$rsync_command = rsync_command( $source_dir, $destination_dir, [ $exclude ] );
+
+	if ( ! EE::exec( $rsync_command ) ) {
+		EE::error( 'Unable to sync files.' );
+	}
+}
+
+function rsync_command( string $source, string $destination, array $options = [] ) {
+	$ssh_command   = 'ssh -t -i ' . get_ssh_key_path();
+	$extra_options = implode( ' ', $options );
+
+	return 'rsync -azh --delete-after --ignore-errors ' . $extra_options . ' -e "' . $ssh_command . '" ' . $source . ' ' . $destination;
+}
+
+function check_site_access( Site $source_site, Site $destination_site, $assoc_args ) {
+	EE::log( 'Checking access to both sites' );
+
+	$source_site->ensure_ssh_success();
+	$source_site->validate_ee_version();
+
+	$destination_site->ensure_ssh_success();
+	$destination_site->validate_ee_version();
+
+	$source_site->set_site_details();
+
+	if ( get_flag_value( $assoc_args, 'overwrite' ) ) {
+		$destination_site->set_site_details();
+	}
+
+}
+
+function get_transfer_details( string $source, string $destination ): array {
+
+	$source_site      = Site::from_location( $source );
+	$destination_site = Site::from_location( $destination );
+
+	if ( ! $source_site->name && ! $destination_site->name ) {
+		EE::error( "No sitename found in source and destination site." );
+	} elseif ( $source_site->ssh_string && $destination_site->ssh_string ) {
+		EE::error( "Both source and destination sites cannot be remote." );
+	} elseif ( ! $source_site->name ) {
+		$source_site->name = $destination_site->name;
+	} elseif ( ! $destination_site->name ) {
+		$destination_site->name = $source_site->name;
+	}
+
+	if ( 'localhost' === $source_site->host && 'localhost' === $destination_site->host && $source_site->name === $destination_site->name ) {
+		EE::error( 'Cannot copy \'' . $source_site->name . '\' on \'' . $source_site->host . '\' to \'' . $destination_site->name . '\' on \'' . $destination_site->host . '\'' );
+	}
+
+	return [ $source_site, $destination_site ];
+}
+
+function get_ssh_key_path() {
+	$user_home = get_user_home_dir( get_current_user() );
+
+	return $user_home . '/.ssh/id_rsa';
```

Comment:
Should not hardcode `~/.ssh/id_rsa`, let it use the key that is there by default. There can be multiple keys on the server and some key other than `~/.ssh/id_rsa` could have access to the other server.

Reply:
If required by the user to use a different key for SSH, then it is up to the user to modify the `~/.ssh/config`.